lib/identity_cache/serializer.rb

```diff
@@ -0,0 +1,83 @@
+module IdentityCache
+  module Serializer # :nodoc:
```

Comment:
Prefer using `private_constant` instead of using a comment to indicate that it is private.  That way it is more obvious that privacy is being worked around in code using the library.  That way you could also avoid using `# :nodoc:` on the methods in this module as well.